src/snowflake/cli/api/project/schemas/project_definition.py

```diff
@@ -45,7 +46,7 @@ class ProjectProperties:
     project_context: Context
 
 
-class _BaseDefinition(UpdatableModel):
+class ProjectDefinition(UpdatableModel):
```

Comment:
I think we should keep the `_` to show that this shouldn't be imported directly. A docstring to explain that this is a base class and not actually a class we can call the initializer on would be good too.

Reply:
I generally agree, but `ProjectDefinition` is also used for type annotations -- it's not only internal to this file. I'll add a docstring to clarify that this is the base class.

Reply:
Sounds good to me. We can keep the invariant that anything with `_` is not intended to be imported outside of the file. How about `ProjectDefinitionBase`?

Reply:
I think signatures using it might be confusing. For example here, it's not the base class that we're returning:
```python
class _CliGlobalContextAccess:
    @property
    def project_definition(self) -> ProjectDefinition:
        return self._manager.project_definition
```

Instead, I created an alias type and kept `_ProjectDefinitionBase` "internal":
```python
ProjectDefinition = _DefinitionV10 | _DefinitionV11 | _DefinitionV20
```